universe/Universe.cpp

```diff
@@ -1641,65 +1641,18 @@ void Universe::ExecuteEffects(std::map<int, Effect::SourcesEffectsTargetsAndCaus
     }
 }
 
-namespace {
-    static const std::string EMPTY_STRING;
-
-    const std::string& GetSpeciesFromObject(std::shared_ptr<const UniverseObject> obj) {
-        switch (obj->ObjectType()) {
-        case OBJ_PLANET: {
-            auto obj_planet = std::static_pointer_cast<const Planet>(obj);
-            return obj_planet->SpeciesName();
-            break;
-        }
-        case OBJ_SHIP: {
-            auto obj_ship = std::static_pointer_cast<const Ship>(obj);
-            return obj_ship->SpeciesName();
-            break;
-        }
-        default:
-            return EMPTY_STRING;
-        }
-    }
-
-    int GetDesignIDFromObject(std::shared_ptr<const UniverseObject> obj) {
-        if (obj->ObjectType() != OBJ_SHIP)
-            return INVALID_DESIGN_ID;
-        auto shp = std::static_pointer_cast<const Ship>(obj);
-        return shp->DesignID();
-    }
-}
-
 void Universe::CountDestructionInStats(int object_id, int source_object_id) {
```

Comment:
I think `CountDestructionInStats` is also called from `ExecuteEffects` for any object that is destroyed by an effect, so this might actually fix a bug where a planet destroyed by an effect could be counted as a ship of its species being destroyed

Reply:
When implementing this I was also a bit confused why destroyed planets would recorded as destroyed ships but good to know that this is a bug.